### PR TITLE
[BoundsSafety][doc] Make it clear that the feature is work-in-progress

### PR DESCRIPTION
**Pull request by rapidsna**
The `-fbounds-safety` feature is not available yet and the implementation is still in progress, which will be guarded by `-fexperimental-bounds-safety`. To avoid the confusion, we're making it more explicit that the feature is not available yet and the document only describes the model.

### CODE REVIEW

**Comment by rapidsna** [comment]
@AaronBallman do you think we should remove the doc from the release notes because it's not ready yet?

File: clang/docs/BoundsSafety.rst [hunk 1]
```diff
@@ -8,6 +8,9 @@
 Overview
 ========
 
+**NOTE:** This is a design document and the feature is not available for users yet.
+Please find :doc:`BoundsSafetyImplPlans` for more details.
```
Comment (AaronBallman) [inline clang/docs/BoundsSafety.rst:12]:
```suggestion
Please see :doc:`BoundsSafetyImplPlans` for more details.
```

**Review by AaronBallman — APPROVED** [review]
Thank you for updating the documentation, the changes LGTM (had one minor suggestion you can take or leave as you'd like). I think it's better for us to have the documentation than not, so I'm fine with this approach of letting people know about the current and future flags.

**Comment by rapidsna** [comment]
@AaronBallman Thanks for your prompt feedback!

**Pull request merged**